packages/as-types/README.md

```diff
@@ -1,72 +1,85 @@
 
-This section contains a description of the classes available in this package.
+- Serialization and Deserialization methods for arrays, staticArrays, booleans, strings and numbers
 
-#### Result
+The complete documentation of all available functions and objects is here:
 
-This class provides a wrapper around a type and an error.
+- [`as-types documentation`](https://as-types.docs.massa.net)
 
-It will allow you to check the returns of various methods available in this package.
 
-Available class methods :
+## Install
 
-- `isOk`
-- `isErr`
-- `expect`
-Checks that the result is okay and panic if not.
-- `unwrap`
-Get the value inside Result object. Panic if error.
+Packages are independant you can choose to install what you need
 
-#### Currency
+```sh
+npm i --save-dev @massalabs/as-types
+```
 
-This class allows for the creation of a currency object with ease.
-It also provides useful methods to manipulate currencies objects, such as :
+## Development guide
 
-- `equals`
-- `notEqual`
-- `toArgs`
-- `fromArgs`
+### Build
 
-```typescript
-const dollar = new Currency("dollar", 2);
+```plain
+npm run build
+```
+
+### Code linting and formatting
+
+```plain
+npm run fmt
 ```
 
-#### Amount
+### Test
+
+```plain
+npm run test
+```
 
-This class allows for the creation of an amount object with ease.
-It also provides useful methods to manipulate amounts objects, such as :
+### Generate doc
+From the root folder :
+```plain
+npm run doc
+```
 
-- `equals`
-- `notEqual`
-- `lessThan`
-- `toArgs`
-- `fromArgs`
+## Usage
+After installing *as-types*, you can import the object classes and functions that you need in your AssemblyScript file.
 
+For example, to use the "args" object, you can import and use it like this:
 ```typescript
-const dollar = new Currency("dollar", 2);
-const price = new Amount(1034, dollar);
+import { Args, i32ToBytes } from  '@massalabs/as-types';
+
+// This main function is called automatically when the smart contract is executed by the blockchain.
+//the argument args contains the serialized values n and k as i32
+export function main(binaryArgs: StaticArray<u8>): StaticArray<u8>{
+	const args = new  Args(binaryArgs);
+	const  n = args.nextI32();
+	const  k = args.nextI32();
+	return i32ToBytes(a + b)
+}
 ```
+## Contributing
+We welcome contributions from the community!
 
-Note that if you try to add or substract amounts that are defined in different types of currencies, it will result in an error.
+If you would like to contribute to Massa-as-sdk, please read the [CONTRIBUTING file](CONTRIBUTING.md).
 
-#### Argument
+## License
+Massa-as-sdk is released under the [MIT License](LICENSE).
 
-This class is very important to write smart contracts.
-It allows for the serialization/deserialization of a regular type (int, float, string), into a binary format.
+## Powered By
+Massa-as-sdk is developed with love by MassaLabs and powered by a variety of [open-source projects](powered-by.md).
```

Comment:
Bad copy/paste ?

Reply:
it is good, I haven't wrote it yet

Reply:
Wdym ? I'm talking about the fact that you are describing 'Massa-as-sdk' here, but this is 'as-types' package

Reply:
my bad, i thought you were speacking about the link